be/src/util/compression/compression_context_pool.h

```diff
@@ -114,14 +114,19 @@ class CompressionContextPool {
 
 private:
     void add(InternalRef ptr) {
+        // Use explicit producer token to avoid the overhead of too many sub-queues
+        static thread_local std::unique_ptr<::moodycamel::ProducerToken> producer_token;
+        if (producer_token == nullptr) {
+            producer_token = std::make_unique<::moodycamel::ProducerToken>(_ctx_resources);
+        }
```

Comment:
it's not thread-safe? could change it to the following:
`static thread_local ::moodycamel::ProducerToken producer_token(_ctx_resources);`

Reply:
Why it's not thread safe, producer_token is thread local.

Reply:
Oh, it's thrad-safe, just a refactor

Reply:
done